CDP4WebServices.API/Services/Operations/OperationProcessor.cs

```diff
@@ -392,6 +397,12 @@ internal void ValidateUpdateOperations(CdpPostOperation operation)
             {
                 throw new InvalidOperationException("FileRevisions updates are not allowed");
             }
+
+            //verify that a participant has a valid domain
```

Comment:
I would not check this here, since we cannot check at this point that the provided DomainOfExpertise is one the ones that is allowed: as per 10-25:
```
Note: The selectedDomain must be one from the set of DomainOfExpertise specified in the domain property of this Participant.
```

We need a side-effect to do this and this should not be done here (we are basically updating the `Participant` instance).